crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -645,18 +641,24 @@ fn handle_deploy_result(
 
 fn serialize_call_trace(call_trace: &CallTrace) -> Vec<Felt252> {
     let mut output = vec![];
-    output.append(&mut serialize_call_entry_point(&call_trace.entry_point));
+
+    serialize_call_trace_inner(call_trace, &mut output);
+
+    output
+}
+
+// append all to one output Vec instead of allocating new one for each nested call
+fn serialize_call_trace_inner(call_trace: &CallTrace, output: &mut Vec<Felt252>) {
+    serialize_call_entry_point(&call_trace.entry_point, output);
```

Comment:
Do we really need this `inner` function? We could make the main one take output and write into it

Reply:
In fact we don't, good point.